lib/iris/tests/unit/coords/test_Cell.py

```diff
@@ -87,29 +86,31 @@ def test_len_1_numpy_array(self):
 
 class Test___eq__(tests.IrisTest):
     def test_datetimelike(self):
-        # Check that cell equality works with objects with a "timetuple".
-        dt = mock.Mock(timetuple=mock.Mock())
-        cell = mock.MagicMock(
-            spec=Cell, point=datetime.datetime(2010, 3, 21), bound=None
+        # Check that cell equality works with different datetime objects
+        # using the same calendar
+        point = cftime.datetime(2010, 1, 1, calendar="gregorian")
+        cell = Cell(
+            datetime.datetime(2010, 1, 1),
+            bound=[
+                datetime.datetime(2010, 1, 1),
+                datetime.datetime(2011, 1, 1),
+            ],
         )
-        _ = cell == dt
-        cell.__eq__.assert_called_once_with(dt)
+        self.assertEqual(cell, point)
 
     def test_datetimelike_bounded_cell(self):
-        # Check that equality with a datetime-like bounded cell
-        # raises an error. This is not supported as it
-        # depends on the calendar which is not always known from
-        # the datetime-like bound objects.
-        other = mock.Mock(timetuple=mock.Mock())
+        # Check that equality with a cell with a different calendar
+        # raises an error. This is not supported
+        point = cftime.datetime(2010, 1, 1, calendar="360_day")
         cell = Cell(
-            point=object(),
+            datetime.datetime(2010, 1, 1),
             bound=[
-                mock.Mock(timetuple=mock.Mock()),
-                mock.Mock(timetuple=mock.Mock()),
+                datetime.datetime(2010, 1, 1),
+                datetime.datetime(2011, 1, 1),
             ],
         )
-        with self.assertRaisesRegex(TypeError, "bounded region for datetime"):
-            cell == other
+        with self.assertRaisesRegex(TypeError, "different calendars"):
+            cell == point
```

Comment:
This is testing for something new - it deserves to be in a new test method.

`test_datetimelike_bounded_cell()` should instead be [how you originally wrote `test_datetimelike()`](https://github.com/SciTools/iris/blob/c48829aab5978ef9c60259d69b1b95cddfed3c68/lib/iris/tests/unit/coords/test_Cell.py#L91C17-L99).

Reply:
Done